clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -222,6 +225,17 @@ void CodeGenModule::createCUDARuntime() {
   CUDARuntime.reset(CreateNVCUDARuntime(*this));
 }
 
+void CodeGenModule::createSYCLRuntime() {
```

Comment:
Rather than testing everywhere I would think a 'noop' SYCL Runtime would be useful.  That way you don't have to worry about this assert.

Reply:
I'm just following existing design for language-specific "runtimes" here. Runtimes are created as needed. If no language-specific support is requested on the command line, then no runtime is created.